File: Data Structures & Algorithms/longest-increasing-subsequence/submission-2.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        vector<int> temp;
        for(int num:nums){
            if(temp.empty() || num > temp.back()){
                temp.push_back(num);
            }
            else{
                int idx = lower_bound(temp.begin(),temp.end(),num) - temp.begin();
                temp[idx] =num;
            }
        }
        return temp.size();
    }
};

File: Data Structures & Algorithms/missing-number/submission-2.java
class Solution {
    public int missingNumber(int[] nums) {
       int ans = 0;
       int st = 0;
       for(int num:nums){
        ans ^= num;
        ans ^= st;
        st++;
       } 
       ans ^= st;
       return ans;
    }
}
